File: angularapp/src/app/shared/contracts/responses/item.ts
import { IEntityResponse } from './IEntityResponse';

interface IItemResponse extends IEntityResponse {}

export interface BasicItemResponse extends IItemResponse {
  itemId: number;
  name: string;
  currently: number;
  buyPrice: number;
  firstBid: number;
  numBids: number;
  started: string;
  ends: string;
  active: boolean;
  description: string;
  sellerId: number;
  imageGuid: string;
}

export interface ExtendedItemInfo extends BasicItemResponse {
  image: {
    src: string;
    isLoading: boolean;
  };
  auctionStarted: string;
  auctionEnds: string;
}

File: webapi/Models/Item.cs
﻿using System.ComponentModel.DataAnnotations;

namespace webapi.Models
{
    public class Item : IModel
    {
        [Key]
        public long ItemId { get; init; }

        public string Name { get; set; } = string.Empty;

        public decimal Currently { get; set; }

        public decimal? BuyPrice { get; set; }

        public decimal FirstBid { get; set; }

        public int NumBids { get; set; }

        public DateTime? Started { get; set; }

        public DateTime? Ends { get; set; }

        public bool Active { get; set; }

        public string Description { get; set; } = string.Empty;

        public int SellerId { get; init; }

        public string? ImageGuid { get; set; }


        public ICollection<Category> Categories { get; set; } = new List<Category>();

        public ICollection<Bid> Bids { get; set; } = new List<Bid>();

        public User Seller { get; set; } = new();
    }

}
